### PR TITLE
add anchor links to h* elements and make the # links show the correct…

### PR DESCRIPTION
**Pull request by chregu**
Makes it easier possible to share links like
https//rokka.io/documentation/references/operations.html#autorotate

and it also prevents from hiding the top 70px behind the bar, due to some scroll (but crude) scroll magic

### CODE REVIEW

File: source-assets/scripts/rokka.js [hunk 1]
```diff
@@ -5,6 +5,19 @@ window.addEventListener('DOMContentLoaded', function() {
...
 
+window.addEventListener('load', function() {
+  if (window.location.hash) {
+    window.scrollBy(0,-70);
+  }
+}, false);
+
+
+window.addEventListener('hashchange', function() {
+  console.log("hello");
```
Comment (michellesanver) [inline source-assets/scripts/rokka.js]: Hai =)

File: src/SculpinTools/AddLinksToId.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        if ( !empty($content)) {
+            $html5 = new HTML5();
+            $dom = $html5->loadHTML($content);
+            $xpath = new \DOMXPath($dom);
+            // search for all elements with an id attribute
+            $nodes = $xpath->query("//*[@id]");
+            foreach ($nodes as $node) {
+                /** @var \DOMElement $node */
+                // only do h1 - h6
+                switch ($node->nodeName) {
```
Comment (michellesanver) [inline src/SculpinTools/AddLinksToId.php]: I would rather have a constant of nodes you do this for. like ... = ['h1', 'h2'... ] and then check if it's in the array. :)

File: source-assets/scripts/rokka.js (hunk 1 above)
Comment (chregu) [reply source-assets/scripts/rokka.js]: oops. fixed ;)

File: src/SculpinTools/AddLinksToId.php (hunk 2 above)
Comment (chregu) [reply src/SculpinTools/AddLinksToId.php]: also fixed

**Pull request merged**